app/goprotoc/codegen.go

```diff
@@ -293,7 +294,18 @@ func writeArchiveResult(fileName string, includeManifest bool, files map[string]
 		}
 	}
 
-	for name, data := range files {
+	fileNames := make([]string, 0, len(files))
+	for name := range files {
+		fileNames = append(fileNames, name)
+	}
+	sort.Strings(fileNames)
+
+	for _, name := range fileNames {
+		data, ok := files[name]
+		if !ok {
```

Comment:
Say what? I'm not a fan of known-useless checks: it's not that it shouldn't happen -- it's that it _can't_.

In fact, if it _did_ happen (like via someone changing the code above in a bad way), I'd prefer letting it fall-through, which will result in a nil-dereference panic, which seems totally appropriate for something like this, which would clearly be a programmer bug.

Reply:
I am of the opinion that panics should never happen if you can prevent them. I treat panics as the equivalent of seg faults - in the case here, we do know what to do if this happened, it's a system error, so we handle it

Reply:
For the sake of avoiding the back and forth, I just did it your way :-)